modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -40,84 +43,92 @@ public interface TxManager extends IgniteComponent {
      * @return {@code True} if a state was changed.
      */
-    boolean changeState(Timestamp ts, @Nullable TxState before, TxState after);
+    boolean changeState(UUID txId, @Nullable TxState before, TxState after);
 
     /**
      * Forgets the transaction state. Intended for cleanup.
      *
-     * @param ts The timestamp.
+     * @param txId Transaction id.
      */
-    void forget(Timestamp ts);
+    void forget(UUID txId);
 
     /**
      * Commits a transaction.
      *
-     * @param ts The timestamp.
+     * @param txId Transaction id.
      * @return The future.
      */
-    CompletableFuture<Void> commitAsync(Timestamp ts);
+    CompletableFuture<Void> commitAsync(UUID txId);
 
     /**
      * Aborts a transaction.
      *
-     * @param ts The timestamp.
+     * @param txId Transaction id.
      * @return The future.
      */
-    CompletableFuture<Void> rollbackAsync(Timestamp ts);
+    CompletableFuture<Void> rollbackAsync(UUID txId);
 
     /**
      * Acqures a write lock.
      *
-     * @param lockId  Table ID.
+     * @param lockId Table ID.
      * @param keyData The key data.
-     * @param ts      The timestamp.
+     * @param txId Transaction id.
      * @return The future.
      * @throws LockException When a lock can't be taken due to possible deadlock.
      */
-    public CompletableFuture<Void> writeLock(IgniteUuid lockId, ByteBuffer keyData, Timestamp ts);
+    public CompletableFuture<Void> writeLock(IgniteUuid lockId, byte[] row, ByteBuffer keyData, UUID txId);
 
     /**
      * Acqures a read lock.
      *
-     * @param lockId  Lock id.
+     * @param lockId Lock id.
      * @param keyData The key data.
-     * @param ts      The timestamp.
+     * @param txId Transaction id.
      * @return The future.
      * @throws LockException When a lock can't be taken due to possible deadlock.
      */
-    public CompletableFuture<Void> readLock(IgniteUuid lockId, ByteBuffer keyData, Timestamp ts);
+    public CompletableFuture<Void> readLock(IgniteUuid lockId, byte[] row, ByteBuffer keyData, UUID txId);
 
     /**
      * Returns a transaction state or starts a new in the PENDING state.
      *
-     * @param ts The timestamp.
+     * @param txId Transaction id.
      * @return @{code null} if a transaction was created, or a current state.
      */
     @Nullable
-    TxState getOrCreateTransaction(Timestamp ts);
+    TxState getOrCreateTransaction(UUID txId);
 
     /**
      * Finishes a dependant remote transactions.
      *
-     * @param ts     The timestamp.
-     * @param addr   The address.
+     * @param txId Transaction id.
+     * @param addr The address.
      * @param commit {@code True} if a commit requested.
      * @param groups Enlisted partition groups.
      */
-    CompletableFuture<Void> finishRemote(NetworkAddress addr, Timestamp ts, boolean commit, Set<String> groups);
+    CompletableFuture<Void> finishRemote(NetworkAddress addr, UUID txId, boolean commit, Set<String> groups);
```

Comment:
For consistency purposes let txId be the last parameter of the method.

Reply:
OK. I fixed it.